locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```diff
@@ -72,9 +74,17 @@
      */
     private int[] inboundPorts;
 
+    private IpProtocol ipProtocol;
+    private RangeSet<Integer> portRanges;
```

Comment:
I think to hold on this syntax for now and revert to your way.
`org.jclouds.net.domain.IpPermission` has much more fields than just protocol and ports.
The most accurate syntax is to use json representation of `IpPermission` but I'd like to keep it now, until someone else asks.